python/cuspatial/cuspatial/tests/test_contains.py

```diff
@@ -0,0 +1,363 @@
+    for detailed examples."""
+)
+def test_one_polygon_with_hole_one_linestring_crossing_it(
+    linestring_generator,
+):
+    gpdlinestring = gpd.GeoSeries(LineString([[0.5, 2.0], [3.5, 2.0]]))
+    gpdpolygon = gpd.GeoSeries(
+        Polygon(
+            (
+                [0, 0],
+                [0, 4],
+                [4, 4],
+                [4, 0],
+                [0, 0],
+            ),
+            [
+                (
+                    [1, 1],
+                    [1, 3],
+                    [3, 3],
+                    [3, 1],
+                    [1, 1],
+                )
+            ],
+        )
+    )
+    linestring = cuspatial.from_geopandas(gpdlinestring)
+    polygons = cuspatial.from_geopandas(gpdpolygon)
+    expected = gpdpolygon.contains(gpdlinestring).values
+    got = polygons.contains(linestring).values_host
+    assert (got == expected).all()
+
+
+@pytest.mark.xfail(
+    reason="""These points are on edges of their corresponding polygons.
+    Because of floating point error, they can have inconsistent results with
+    GeoPandas. In this case, GeoPandas results are incorrect due to their
+    geometry engine. It is possible that implementing `.contains_properly`
+    would correct this error. These boundary cases conflict with GeoPandas
+    results because they implement `contains` and we implement
+    `contains_properly`. Detailed documentation of this in
+    https://docs.google.com/document/d/1akxcRcUVK-qv5puK-mSTiKDKUS6UlR6Ccnr9_bJ4fi8/edit?pli=1#
+    hasn't been completed yet."""
+)
+@pytest.mark.parametrize(
+    "point, polygon, expects",
+    [
+        [
+            Point([0.6, 0.06]),
+            Polygon([[0, 0], [10, 1], [1, 1], [0, 0]]),
+            False,
+        ],
+        [
+            Point([3.333, 1.111]),
+            Polygon([[6, 2], [3, 1], [3, 4], [6, 2]]),
+            True,
+        ],
+        [Point([3.33, 1.11]), Polygon([[6, 2], [3, 1], [3, 4], [6, 2]]), True],
+    ],
+)
+def test_float_precision_limits_failures(point, polygon, expects):
+    gpdpoint = gpd.GeoSeries(point)
+    gpdpolygon = gpd.GeoSeries(polygon)
+    point = cuspatial.from_geopandas(gpdpoint)
+    polygon = cuspatial.from_geopandas(gpdpolygon)
+    expected = gpdpolygon.contains(gpdpoint).values
+    got = polygon.contains(point).values_host
+    assert got == expected
+    assert got.values_host[0] == expects
+
+
+@pytest.mark.parametrize(
+    "point, polygon, expects",
+    [
+        # unique failure cases identified by @mharris
+        [
+            Point([0.66, 0.006]),
+            Polygon([[0, 0], [10, 1], [1, 1], [0, 0]]),
+            False,
+        ],
+        [
+            Point([0.666, 0.0006]),
+            Polygon([[0, 0], [10, 1], [1, 1], [0, 0]]),
+            False,
+        ],
+        [Point([3.3, 1.1]), Polygon([[6, 2], [3, 1], [3, 4], [6, 2]]), True],
+    ],
+)
+def test_float_precision_limits(point, polygon, expects):
```

Comment:
... And add a 1 liner comment here:
```suggestion
def test_float_precision_limits(point, polygon, expects):
    """Corner case to test point on edges with floating point precision limits."""
```